src/code/z_play.c

```diff
@@ -1,59 +1,1462 @@
+                if (1 && HREG(63) != 0) {
+                    LOG_NUM("1", 1, "../z_play.c", 3721);
+                }
+
+                KaleidoScopeCall_Update(globalCtx);
+            } else if (globalCtx->unk_10A20 != 0) {
+                if (1 && HREG(63) != 0) {
+                    LOG_NUM("1", 1, "../z_play.c", 3727);
+                }
+
+                func_801104C8(globalCtx);
+            } else {
+                if (1 && HREG(63) != 0) {
+                    LOG_NUM("1", 1, "../z_play.c", 3733);
+                }
+
+                func_8010F6F0(globalCtx);
+            }
+
+            if (1 && HREG(63) != 0) {
+                LOG_NUM("1", 1, "../z_play.c", 3737);
+            }
+
+            if (1 && HREG(63) != 0) {
+                LOG_NUM("1", 1, "../z_play.c", 3742);
+            }
+
+            Interface_Update(globalCtx);
+
+            if (1 && HREG(63) != 0) {
+                LOG_NUM("1", 1, "../z_play.c", 3765);
+            }
+
+            func_800A390C(globalCtx, &globalCtx->animationCtx);
+
+            if (1 && HREG(63) != 0) {
+                LOG_NUM("1", 1, "../z_play.c", 3771);
+            }
+
+            func_8006BA30(globalCtx);
+
+            if (1 && HREG(63) != 0) {
+                LOG_NUM("1", 1, "../z_play.c", 3777);
+            }
+
+            func_800B39B8(R_UPDATE_RATE);
+
+            if (1 && HREG(63) != 0) {
+                LOG_NUM("1", 1, "../z_play.c", 3783);
+            }
+
+            func_800B3568(&globalCtx->sub_1241C, R_UPDATE_RATE);
+        }
+    }
+
+    if (1 && HREG(63) != 0) {
+        LOG_NUM("1", 1, "../z_play.c", 3799);
+    }
+
+    if (1 && HREG(63) != 0) {
+        LOG_NUM("1", 1, "../z_play.c", 3801);
+    }
+
+    if ((sp80 == 0) || (D_8011D394 != 0)) {
+        s32 i; // 0x54
+        s32 camIdx;
+        Vec3s sp48;
+
+        globalCtx->nextCamera = globalCtx->activeCamera;
+
+        if (1 && HREG(63) != 0) {
+            LOG_NUM("1", 1, "../z_play.c", 3806);
+        }
+
+        for (i = 0, camIdx = globalCtx->nextCamera; i < 4; i++) {
+            if ((i != camIdx) && (globalCtx->cameraPtrs[i] != NULL)) {
+                if (1 && HREG(63) != 0) {
+                    LOG_NUM("1", 1, "../z_play.c", 3809);
+                }
+
+                func_800591EC(&sp48, globalCtx->cameraPtrs[i]);
+                camIdx = globalCtx->nextCamera;
+            }
+        }
+
+        func_800591EC(&sp48, globalCtx->cameraPtrs[camIdx]);
+
+        if (1 && HREG(63) != 0) {
+            LOG_NUM("1", 1, "../z_play.c", 3814);
+        }
+    }
+
+    if (1 && HREG(63) != 0) {
+        LOG_NUM("1", 1, "../z_play.c", 3816);
+    }
+
+    func_80070C24(globalCtx, &globalCtx->envCtx, &globalCtx->lightCtx, &globalCtx->pauseCtx, &globalCtx->msgCtx,
+                  &globalCtx->unk_10A20, globalCtx->state.gfxCtx);
+}
+#else
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/Gameplay_Update.s")
+#endif
+
+void Gameplay_DrawOverlay(GlobalContext* globalCtx) {
```

Comment:
I know this name comes from the fact that the dlist buffer is called overlay. Can be hella confusing at first glance though. I would say DrawHUD but yeah kaleidoscope isnt a HUD... oh well I dont have any better suggestions.

Reply:
Yeah "overlay" is confusing but that's kinda already the case due to the conflict with terms we didn't chose... Something like DrawOverlayElements might be less confusing and still pretty descriptive tho.
I think it's important to keep in mind the scope/context of this function. It's only ever called in Gameplay_Draw, and this system/file doesn't deal with actual overlays in any way, not that it would make sense to have a function that "draws an overlay" anyway since an overlay is really just a file format.

Reply:
Yeah absolutely, it makes a ton of sense with context considered. Was just a thought I had at first glance. I do like DrawOverlayElements fwiw

Reply:
I don't think we should change it just because of DLFs. DLF overlays are data storage, and should be considered to be "hidden behind the scenes". the game engine isn't aware of dlfs.

Reply:
Yeah, I renamed it to Gameplay_DrawOverlayElements. I think it should be fine considering the context.